### ISSUE TITLE
`ValueKind` is a `Value`, not a `Kind`

### ISSUE THREAD

**Issue by epage** [rust-cli/config-rs#590]
`Kind`s in Rust API design are basically variant discriminants.  This stores the underlying value.

**Comment by polarathene** [comment]
If refactoring, this PR is likely related: https://github.com/rust-cli/config-rs/pull/472

It's goal was to centralize conversion from config.

**Comment by epage** [comment]
imo several of those commits seem stand on their own enough that splitting them out into individual PRs would be a big help for moving getting those merged more quickly.

**Comment by polarathene** [comment]
Yes I understand, I still would like to pick it back up, especially with project maintenance being restored.

I just overcommitted myself over the past year, burned out and playing catchup. December is probably the earliest I'd have time to allocate to revisiting my contributions here.

**Comment by epage** [comment]
Thats perfectly fine!  I'm playing catch up with issues and PRs from months ago :)
